experimental/rsocket/RSocketErrors.h

```diff
@@ -0,0 +1,133 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+
+namespace rsocket {
+
+/*
+ * Error Codes from
+ * https://github.com/ReactiveSocket/reactivesocket/blob/master/Protocol.md#error-codes
+ */
+class RSocketError : public std::runtime_error {
+ public:
+  explicit RSocketError(const std::string& s)
+      : std::runtime_error(std::move(s)) {}
+  explicit RSocketError(const char* s) : std::runtime_error(std::move(s)) {}
+  /**
+   * Get the error code for inclusion in an RSocket ERROR frame as per
+   * https://github.com/ReactiveSocket/reactivesocket/blob/master/Protocol.md#error-codes
+   * @return
+   */
+  virtual int getErrorCode() = 0;
+  /**
+   * Get the string name of the error as per
+   * https://github.com/ReactiveSocket/reactivesocket/blob/master/Protocol.md#error-codes
+   * @return
+   */
+  virtual const std::string getErrorString() = 0;
```

Comment:
This is different than what(). My intent here was to specifically expose just the text of the spec, and that what() would expose the error message passed in by the constructor.